### DISCUSSION TITLE
Have an "Automatic" option for ModalSize

### DISCUSSION THREAD

**Discussion by NonoSparc** [Blazored/Modal#443] [Ideas]
In versions previous to 7.0, the size of the modal was adaptative to the Blazor component shown. It was not necessary to set the size at the time of the Modal.Show call.

Could it be possible to have this behavior as a option in the ModalSize enum? Or maybe have the defined size (or default if not provided) be a "minimum" size only?

**Comment by chrissainty** [comment]
Yes. This is a good idea nd I should have thought of it during the v7 work. I'll get this converted to an issue.

**Comment by NonoSparc** [comment]
I’m fine with using ModalSize.Custom option, but then SizeCustomClass should be optional.

Currently you get an error if you omit it.

De : Chris Sainty ***@***.***>
Envoyé : mercredi 12 octobre 2022 14:44
À : Blazored/Modal ***@***.***>
Cc : Bruno Crapoulet ***@***.***>; Mention ***@***.***>
Objet : Re: [Blazored/Modal] Have an "Automatic" option for ModalSize (Discussion #443)

CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.

Having thought about this @NonoSparc<https://github.com/NonoSparc>, ModalSize.Custom option should provide the desired behaviour.
